src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs

```diff
@@ -77,8 +77,12 @@ internal static TAutoHistory AutoHistory<TAutoHistory>(this EntityEntry entry, F
                     {
                         if (prop.IsModified)
                         {
-                            bef[prop.Metadata.Name] = prop.OriginalValue != null
-                            ? JToken.FromObject(prop.OriginalValue, jsonSerializer)
+                            //bef[prop.Metadata.Name] = prop.OriginalValue != null
+                            //? JToken.FromObject(prop.OriginalValue, jsonSerializer)
+                            //: JValue.CreateNull();
+                            var originalValue = entry.GetDatabaseValues().GetValue<object>(prop.Metadata.Name);
```

Comment:
Here we has more database roundtrip even if the `prop.OriginalValue != null`. There have some way to know the entry if attach or fetch from database? If the we know that, we can invoke `entry.GetDatabaseValues()` only in attach case.

@threax Do you  think so?

Reply:
I will try to find better option but this is what I found for now.  If you have an option to avoid DB trip, please do let me know. Thanks!